### PR TITLE
https://contest.yandex.ru/contest/38500/run-report/69037214/

### PR DESCRIPTION
**Pull request by havo2001**

### CODE REVIEW

File: task1.cpp [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+            continue;
+        } else {
+            if (j == 0) {
+                prefix[i] = 0;
+                continue;
+            } else {
+                j = prefix[j - 1];
+            }
+        }
+    }
```
Comment (DzhangirliEmil) [inline task1.cpp]: counting the prefix vector should be done in a separate function

File: task1.cpp [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    int i = 0;
+    j = 0;
+    while (i < size_str) {
+        if (str[i] == pattern[j]) {
+            i++;
```
Comment (DzhangirliEmil) [inline task1.cpp:36]: ++i

**Review by DzhangirliEmil — CHANGES_REQUESTED** [review]
place a valid link to your solution in the contest

File: task1.cpp (hunk 1 above)
Comment (havo2001) [reply task1.cpp:25]: I tried but if i calculated in seperate function, i got TL

File: task1.cpp (hunk 2 above)
Comment (havo2001) [reply task1.cpp:36]: i have already fixed

File: task1.cpp (hunk 1 above)
Comment (havo2001) [reply task1.cpp]: I have fixed it

**Comment by havo2001** [comment]
https://contest.yandex.ru/contest/38500/run-report/69037214/

**Pull request merged**